pkg/storage/fs/cback/utilities.go

```diff
@@ -0,0 +1,360 @@
+
+	if err != nil {
+		return nil, err
+	}
+
+	retObject := fsReturn{
+		Path:   source + "/" + snapID + strings.TrimPrefix(responseObject.Name, source),
+		Type:   m,
+		Mtime:  uint64(responseObject.Mtime),
+		Size:   responseObject.Size,
+		Detail: responseObject.Detail,
+	}
+
+	return &retObject, nil
+}
+
+func (fs *cback) fileSystem(backupID int, snapID, userName, path, source string) ([]fsReturn, error) {
+
+	url := fs.conf.APIURL + "/backups/" + strconv.Itoa(backupID) + "/snapshots/" + snapID + "/" + path + "?content=true"
+	requestType := "OPTIONS"
+
+	responseData, err := fs.getRequest(userName, url, requestType, nil)
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer responseData.Close()
+
+	/*Unmarshalling the JSON response into the Response struct*/
+	responseObject := []contents{}
+	json.NewDecoder(responseData).Decode(&responseObject)
+
+	resp := make([]fsReturn, len(responseObject))
+
+	for i, response := range responseObject {
+
+		m, err := mapReturn(response.Type)
+
+		if err != nil {
+			return nil, err
+		}
+		/*fmt.Printf("\nName is: %v\n", responseObject[i].Name)
+		fmt.Printf("Type is: %d\n", m)
+		fmt.Printf("Time is: %v", uint64(responseObject[i].Mtime))
+		fmt.Printf("\n")*/
+
+		resp[i].Size = response.Size
+		resp[i].Type = m
+		resp[i].Mtime = uint64(response.Mtime)
+		resp[i].Path = source + "/" + snapID + strings.TrimPrefix(response.Name, source)
+
+	}
+
+	return resp, nil
+}
+
+func (fs *cback) timeConv(timeInput string) (int64, error) {
+	tm, err := time.Parse(time.RFC3339, timeInput)
+
+	if err != nil {
+		return 0, err
+	}
+
+	epoch := tm.Unix()
+	return epoch, nil
+}
+
+func (fs *cback) pathFinder(userName, path string) ([]string, error) {
+	url := fs.conf.APIURL + "/backups/"
+	requestType := "GET"
+	responseData, err := fs.getRequest(userName, url, requestType, nil)
+	matchFound := false
+
+	if err != nil {
+		return nil, err
+	}
+
+	defer responseData.Close()
+
+	/*Unmarshalling the JSON response into the Response struct*/
+	responseObject := []backUpResponse{}
+	json.NewDecoder(responseData).Decode(&responseObject)
+
+	returnString := make([]string, len(responseObject))
+
+	for index, response := range responseObject {
+		if response.Detail != "" {
+			err = errors.New(response.Detail)
+			return nil, err
+		}
+
+		if strings.HasPrefix(response.Source, path) {
+			substr := strings.TrimPrefix(response.Source, path)
+			substr = strings.TrimLeft(substr, "/")
+			temp := strings.Split(substr, "/")
+			returnString[index] = temp[0]
+			matchFound = true
+		}
+	}
+
+	if matchFound {
+		return duplicateRemoval(returnString), nil
+	}
+
+	return nil, errtypes.NotFound("cback: resource not found")
+
+}
+
+func (fs *cback) pathTrimmer(snapshotList []snapshotResponse, resp *backUpResponse) (string, string) {
+
+	var ssID, searchPath string
+	for _, snapshot := range snapshotList {
+
+		if snapshot.ID == resp.Substring {
+			ssID = resp.Substring
+			searchPath = resp.Source
+			break
```

Comment:
return instead of break, is not clear otherwise the flow

Reply:
Discussed with Gianmaria